driver/src/contracts/stablex_contract.rs

```diff
@@ -128,9 +130,14 @@ impl<'a> StableXContract for StableXContractImpl<'a> {
                 prices,
                 token_ids_for_price,
             )
-            .call()
-            .wait()?;
+            .view();
 
+        let builder = if let Some(block_number) = block_number {
```

Comment:
```suggestion
        builder.block = block_numer
```

The builder properties are public, the builder methods are just convenient in most cases.